tests/core/atoms/test_atoms_core.py

```diff
@@ -208,3 +213,18 @@ def test_check_charge_and_spin(os_atoms):
     charge, spin_multiplicity = check_charge_and_spin(atoms, charge=0)
     assert charge == 0
     assert spin_multiplicity == 3
+
+
+def test_perturb():
```

Comment:
Could you add one additional check to make sure that the original `atoms` object is unmodified after calling `perturb`? Thanks!